src/shared_utilities/drivers.jl

```diff
@@ -185,11 +186,110 @@ PrescribedPrecipitation{FT}(liquid_precip) where {FT} =
     PrescribedPrecipitation{FT, typeof(liquid_precip)}(liquid_precip)
 
 """
-    CoupledRadiativeFluxes{FT} <: AbstractRadiativeDrivers{FT}
+    CoupledRadiativeFluxes{
+        FT,
+        F <: Union{Function, Nothing},
+        T,
+    } <: AbstractRadiativeDrivers{FT}
+
+To be used when coupling to an atmosphere model. Either both `θs` and `start_date`
+must be `nothing`, or both must be non-nothing.
 
-To be used when coupling to an atmosphere model.
+During the driver update, cosθs is unchanged if `θs` is `nothing`.
+Otherwise, `θs` recieves the following arguments:
+(time_from_start, `start_date`), and is expected to update return zenith angle at the given time.
+$(DocStringExtensions.FIELDS)
 """
-struct CoupledRadiativeFluxes{FT} <: AbstractRadiativeDrivers{FT} end
+struct CoupledRadiativeFluxes{FT, F <: Union{Function, Nothing}, T} <:
+       AbstractRadiativeDrivers{FT}
+    """Function that fills a climacore field with the zentih angle given the following arguments:
+    (time_from_start, `start_date`)"""
+    θs::F
+    "Start date - the datetime corresponding to t=0 for the simulation"
+    start_date::T
+    function CoupledRadiativeFluxes{FT, F, T}(
+        θs::F,
+        start_date::T,
+    ) where {FT, F, T}
+        (!(isnothing(θs) ⊻ isnothing(start_date))) || error(
```

Comment:
I think this is a little more readable:
```suggestion
        ((isnothing(θs) && isnothing(start_date)) || ((!isnothing(θs) && !isnothing(start_date))) || error(
```